### DISCUSSION TITLE
playlist_osd打开状态下，再次按打开键不会关闭。

### DISCUSSION THREAD

**Discussion by yeezylife** [hooke007/mpv_PlayKit#357] [回收]
“##高级播放列表打开后，此时方向键切换与选中，回车播放，退格键移除，Esc键关闭（超时自动关闭或者再次按打开键也会关闭）”

和input_scripts.conf描述似乎不符合。

我的用shinchiro的最新版mpv，删掉所有其他脚本只留下playlist_osd.lua，mpv.conf里只留一行"no-config"，input.conf里只留一行"Shift+ENTER script-binding playlist_osd/display"测试的。

按Shift+ENTER能正常唤出播放列表，但是想要退出再按Shift+ENTER没有用，只能用esc退出

**Comment by zhongfly** [comment]
https://mpv.io/manual/master/#options-no-config
你用no-config时，input.conf也是会被忽略的，你这么测试没起作用…

**Reply by yeezylife** [reply]
不开no-config也是一样的行为，两种我都测过。。。

**Reply by zhongfly** [reply]
我觉得在mpv.conf用no_config到底有没有作用是个问题。按理说是应该用于命令行的，`mpv --no-config`

**Reply by hooke007** [reply]
config=no 只在初始化时生效，所以是纯命令行选项

**Comment by hooke007** [comment]
这段描述应该是改组前保留的文字给被我遗忘修正了

**Comment by verygoodlee** [comment]
如果想同一键位开启、关闭的话，这样写应该可以解决
`Shift+ENTER script-message playlist_osd show playlist toggle`
[原版playlistmanager](https://github.com/jonniek/mpv-playlistmanager)是支持这种写法，看了下这部分的是代码保留的
https://github.com/hooke007/MPV_lazy/blob/ec01ac4ea2bd58b9a86220aa0aee5a8c98b1ea47/portable_config/scripts/playlist_osd.lua#L589-L620

**Reply by verygoodlee** [reply]
@hooke007 好像不行，原版函数名`showplaylist`你这边改成`playlist_show`了，所以595行要改成`toggle_playlist(playlist_show) `
https://github.com/hooke007/MPV_lazy/blob/ec01ac4ea2bd58b9a86220aa0aee5a8c98b1ea47/portable_config/scripts/playlist_osd.lua#L595
335行最好也改一下`local show = show_function or playlist_show`
https://github.com/hooke007/MPV_lazy/blob/ec01ac4ea2bd58b9a86220aa0aee5a8c98b1ea47/portable_config/scripts/playlist_osd.lua#L335

**Reply by yeezylife** [reply]
按照你说的修改完已经可以啦
